### ISSUE TITLE
[java] LawOfDemeter: false negative with method chain call in Lambda

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4375]
**Affects PMD Version:** 6.52.0

**Rule:**[LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**
The AnonymousClass can be converted into Lambda. The method chain call can be detected in the AnonymousClass case, but can not be detected in the Lambda class. So it may be a false negative.

**Code Sample demonstrating the issue:**
```java
import java.awt.*;
import java.awt.event.*;
import java.util.*;
import javax.swing.*;
import javax.swing.tree.*;

public class Bar {
    private final JTree tree = new JTree();
    public JComponent makeUI() {
        Box box = Box.createVerticalBox();
        box.add(new JButton((ActionEvent e) -> {
            TreeNode root = (TreeNode) tree.getModel().getRoot();    // NOT report 
            visitAll(tree, new TreePath(root), true);
        }));
        box.add(new JButton(new AbstractAction("collapse") {
            @Override
            public void actionPerformed(ActionEvent e) {
                TreeNode root = (TreeNode) tree.getModel().getRoot();    // report LawOfDemeter(method chain calls)
                visitAll(tree, new TreePath(root), false);
            }
        }));
        box.add(Box.createVerticalGlue());
        JPanel p = new JPanel(new BorderLayout());
        p.add(box, BorderLayout.EAST);
        p.add(new JScrollPane(tree));
        return p;
    }
}
```

**Expected outcome:**
```
LawOfDemeter:        Potential violation of Law of Demeter (method chain calls)
```
PMD should report a violation at line ..., but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
I confirm this is working as intended in PMD 7.0.0-rc3. Feel free to update!

**Comment by mohui1999** [comment]
Thanks for your replay!
